### PR TITLE
Add healthcheck URL.

### PR DESCRIPTION
**Pull request by glogiotatidis**
This URL will return 200 OK. To be used with Deis health checks.

### CODE REVIEW

**Comment by pmclanahan** [comment]
Why do we need a special url for this? I thought it'd be better to have a real page checked which would ensure that the whole site is working. The home page even ensures that the DB is working. I'm probably misunderstanding the purpose of this thing.

File: vendor-local/src/legal-docs [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit fdf03d97dff7a53e6331018649c357fa6eea9b9e
+Subproject commit 3e3426f74e477265f3231c9b332cfe169ae35264
```
Comment (pmclanahan) [inline vendor-local/src/legal-docs]: Probably didn't mean to include this.

**Comment by glogiotatidis** [comment]
Valid question. I'm still not convinced that this is needed but I wanted to build a test image to test it, thus the pull request. 

Deis health checks against a relative url e.g. `/` instead of a full url e.g. `http://bedrock-dev.us-west.moz.works` which makes sense. The problem is that we redirect all requests to https and therefore the health check against `/en-US/` returns `301` instead of `200`. 

This PR attempts to solve this by providing a non-https URL to check against.

I'll investigate further and update here.

**Comment by glogiotatidis** [comment]
Docs here: http://docs.deis.io/en/latest/using_deis/config-application/?highlight=health#custom-health-checks

**Comment by pmclanahan** [comment]
Ah! I'd forgotten about the HTTPS 301 even though your PR clearly opts `/200/` out of that specifically. Seems good then.

**Comment by pmclanahan** [comment]
r+ from me then as soon as the submodule change is removed.

**Comment by jgmize** [comment]
There is a convention of using `/healthz` as the endpoint for health checks that we may want to adopt. It's the example endpoint in the [kubernetes docs](http://kubernetes.io/v1.0/docs/user-guide/liveness/README.html), deis will [adopt it internally](https://github.com/deis/deis/issues/4482) as the project switches to k8s in v2, and other large open source projects such as cloud foundry are [adopting it](https://github.com/openshift/origin/blob/b2e97700db09415052d60c02c9edf49bfaf506b3/pkg/cmd/server/origin/handlers.go#L35) as well.

**Comment by glogiotatidis** [comment]
Rebased and tested http://giorgosbedrock.eu-west.moz.works/healthz/

**Comment by glogiotatidis** [comment]
Merge at will

**Comment by glogiotatidis** [comment]
please hold merging this, I want to test another thing.

**Comment by glogiotatidis** [comment]
I tried to enabled checks on my app with

`deis_eu config:set HEALTHCHECK_URL=/healthz/  HEALTHCHECK_INITIAL_DELAY=50 -a giorgosbedrock`

and I get `detail: aborting, app containers failed to respond to health check`

**Comment by jgmize** [comment]
@glogiotatidis I suspected that your HEALTHCHECK_INITIAL_DELAY was too small, due to the time it takes to download the maxmind db, so to test that theory I set HEALTHCHECK_INITIAL_DELAY=300 on bedrock-dev on both clusters and triggered [bedrock pipeline #144](https://ci.us-west.moz.works/view/Bedrock%20Pipeline/?fullscreen=true) by building [bedrock_base_image with parameters TAG=pr/3389/head](https://ci.us-west.moz.works/view/Bedrock/job/bedrock_base_image/144/console).

**Comment by jgmize** [comment]
The deploy was successful and https://bedrock-dev.us-west.moz.works/healthz/ returns OK for me now. We can probably tune HEALTHCHECK_INITIAL_DELAY lower than 300, but I believe this should be fine to merge as-is. Do you agree, @glogiotatidis?

**Comment by glogiotatidis** [comment]
Interesting. I'm wondering whether the healthcheck delay starts the moment the unit files get created *or* when the port opens. I'd expect the later thus the small delay in my setup but it's probably the first. 

My only concern is if Deis will need at least 300 seconds (i.e. 5 minutes) to make a deploy? Otherwise I'm good to merge.

**Comment by jgmize** [comment]
> Interesting. I'm wondering whether the healthcheck delay starts the moment the unit files get created or when the port opens. I'd expect the later thus the small delay in my setup but it's probably the first.

The publisher [listens](https://github.com/deis/deis/blob/v1.10.1/publisher/server/publisher.go#L47) to docker API events on each host, calls [publishContainer](https://github.com/deis/deis/blob/v1.10.1/publisher/server/publisher.go#L64) when the container is created, which does a [health check](https://github.com/deis/deis/blob/v1.10.1/publisher/server/publisher.go#L146) which will [sleep for the configured delay](https://github.com/deis/deis/blob/v1.10.1/publisher/server/publisher.go#L206) before testing the configured url, and only publish to etcd on success. 

> My only concern is if Deis will need at least 300 seconds (i.e. 5 minutes) to make a deploy? Otherwise I'm good to merge.

Yes, I'd like to eliminate this delay as well, but as this is set outside of the code I don't think that's a blocker for merge. I only set it that high to test my theory-- we can experiment with different values and see what works.

**Pull request merged**